### PR TITLE
C2PA-273: Use of generic table implementations

### PR DESCRIPTION
**Pull request by brogdonm**
## Changes in this pull request
_Give a narrative description of what has been changed._

This moves the tables to a generic table implementation. This should set us up to have a generic font trait for handling all the common things a font does.

## Checklist
- [ ] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/asset_handlers/font_io.rs [hunk 1]
```diff
@@ -22,10 +23,14 @@ use std::{
...
-use crate::error::{Error, Result};
+use crate::{
+    error::{Error, Result},
+    CAIRead, CAIReadWrite,
+};
 
-/// Types for supporting fonts in any container.
+impl<T> CAIRead for BufReader<T> where T: Read + Send + Seek {}
 
+/// Types for supporting fonts in any container.
```
Comment (vancejc-mt) [inline sdk/src/asset_handlers/font_io.rs]: Rogue comment?

File: sdk/src/asset_handlers/sfnt_io.rs [hunk 2]
```diff
@@ -179,15 +179,11 @@ mod font_xmp_support {
     /// offers a more generic method to indicate a document ID, instance ID,
     /// and a reference to the a remote manifest.
     #[allow(dead_code)]
-    pub(crate) fn add_reference_as_xmp_to_stream<TSource, TDest>(
-        source: &mut TSource,
-        destination: &mut TDest,
+    pub(crate) fn add_reference_as_xmp_to_stream(
+        source: &mut dyn CAIRead,
```
Comment (vancejc-mt) [inline sdk/src/asset_handlers/sfnt_io.rs:183]: Noob question, but why is this `dyn CAIRead` and not `impl CAIRead`?

**Review by vancejc-mt — APPROVED** [review]
Still works as advertised, definitely looks cleaner, though having to do all the dynamic casting is definitely wonkier than dealing with interfaces (traits != interfaces, will never learn).

Looks good though - just saw maybe a leftover comment? And rust question

File: sdk/src/asset_handlers/sfnt_io.rs (hunk 2 above)
Comment (brogdonm) [reply sdk/src/asset_handlers/sfnt_io.rs:183]: The `impl CAIRead` is basic syntax sugar for generic parameters. And I am not going to claim I fully understand this, but if you use generic parameters in the trait, it cannot "safely" create object. So using the `dyn CAIRead` allows for dispatching execution instead. 

See https://doc.rust-lang.org/reference/items/traits.html#object-safety for more details.

File: sdk/src/asset_handlers/font_io.rs (hunk 1 above)
Comment (brogdonm) [reply sdk/src/asset_handlers/font_io.rs]: Yeah, looks like it from some time in the past. Will remove.

Comment (brogdonm) [reply sdk/src/asset_handlers/font_io.rs]: Fixed and pushed.

**Review by vancejc-mt — APPROVED** [review]
Looks good!

**Pull request merged**